typeclasses/characters.py

```diff
@@ -84,6 +86,101 @@ def at_pre_unpuppet(self):
         if self.nattributes.has('combat_handler'):
             self.ndb.combat_handler.remove_character(self)
 
+    def return_appearance(self, looker):
+        """
+        This formats a description. It is the hook a 'look' command
+        should call.
+        Args:
+            looker (Object): Object doing the looking.
+        """
+        if not looker:
+            return ""
+
+        # get description, build string
+        name = self.get_display_name(looker)
+        per  = self.traits.PER
+
+        # These each do a skill check, but they always pass
+        # if you're looking at yourself
+        knows_race = skill_check(per) or looker == self
+        knows_archetype = skill_check(per) or looker == self
+        knows_health = skill_check(per) or looker == self
+        knows_stamina = skill_check(per) or looker == self
+
+        # this is the base name format - it just colors the name cyan
+        string = "|c%s|n" % name
+
+        # if we're adding race or archetype, add "the" after the name
+        if (knows_race and self.db.archetype) or (knows_archetype and self.db.archetype):
```

Comment:
You could add these conditionals to the initialization, since if the character doesn’t have a race, I can’t know what race they are. You’re also checking `archetype` instead of `race` in the first condition.